src/platform/apollolake/include/platform/drivers/interrupt.h

```diff
@@ -11,32 +11,55 @@
 #ifndef __PLATFORM_DRIVERS_INTERRUPT_H__
 #define __PLATFORM_DRIVERS_INTERRUPT_H__
 
+#include <config.h>
```

Comment:
@tlauda I would advise on not touching the headers. It doesn't save that much space for modifying the headers, meanwhile the code become very unpleasant to read.

Reply:
@dbaluta We need to touch the headers if we want to do it this way. Compiling out certain interrupt level definitions is serving as a guard. If in the future there will be need to use some additional interrupt level we will notice the need to enable this config option right away. Without it there will be no code for interrupt vector and we'll get exception. Note that currently every irq level requires 4 kB of stack. On cAVS we are getting back 12 kB with this patch, not mentioning the code for handling.

Reply:
Just spreading those #ifdefs all around the code makes it very hard to read. So, I'm ok with ifdefs to compile out data and code, but for most the headers the defines are harmless.

Reply:
Is the argument here to cause compiler errors if you try to use something unavailable in your config? Instead of e.g. linker errors (for functions) or silent usage of something unsupported (in case of anything else)?

Reply:
@lgirdwood @dbaluta I think I didn't make myself clear and only @paulstelian97 understood the issue here. Without #ifdefs in platform/drivers/interrupt.h we will face runtime exceptions. Let's take an example: someone decides to change PLATFORM_SCHEDULE_IRQ to IRQ_X which is level 4. CONFIG_INTERRUPT_LEVEL_4 is not set for this platform. Code will compile fine, but DSP will panic as soon as irq lvl 4 will be triggered. However with #ifdefs in headers we will get compile error as soon as user tries to use any irq lvl 4.

Reply:
@tlauda I see your point. The PR is really good, we need to find a way to not harm code readability. 
In your example, PLATFORM_SCHEDULE_IRQ maybe can be set from Kconfig and it will select as a dependency the correct interrupt level. I'm not against your change per se, lets spend some time to find a cleaner solution.

Reply:
@dbaluta Scheduler is just an example. There are infinite possibilities where someone can try to use different interrupts. The easiest way is to just cause compile error. More #ifdefs in headers (especially platform specific) is a small price to pay for early error detection.

Reply:
@tlauda ok, so for me I want to preseve readability and that may mean moving all macros pertaining to IRQ level X to a single block (i.e. we only have one ifdef IRQ_LEVEL_2 in the header, one for 3, one for 4 etc).

Reply:
@lgirdwood Done.